sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3288,7 +3288,15 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
         throw QueryParsingErrors.cannotCleanReservedTablePropertyError(
           PROP_EXTERNAL, ctx, "please use CREATE EXTERNAL TABLE")
       case (PROP_EXTERNAL, _) => false
-      case _ => true
+      // It's safe to set whatever table comment, so we don't make it a reserved table property.
+      case (PROP_COMMENT, _) => true
+      case (k, _) =>
```

Comment:
I think it is duplicate with case before. Just leave code you added is simple and clear.

Reply:
We need the code above to throw errors with a better error message, which gives suggestions like `please use CREATE EXTERNAL TABLE`

Reply:
That's where I got confused too. If we want to keep the code above to issue a better error message, why not adding something in parallel such as 
```
case (PROP_IS_MANAGED_LOCATION, _) if !legacyOn =>
    throw QueryParsingErrors.cannotCleanReservedTablePropertyError(
       PROP_IS_MANAGED_LOCATION, ctx, "xxxxxx")
case (PROP_IS_MANAGED_LOCATION, _) =>
    false
```

Reply:
We may add more reserved properties in the future that do not have alternatives such as `CREATE EXTERNAL TABLE`, so I wrote the code this way to ease the work of adding new reserved properties.

Reply:
Got it. Thanks for the explanation.